### ISSUE TITLE
SNOW-654658: SnowparkFetchDataException when using to_pandas after v0.9.0 upgrade

### ISSUE THREAD

**Issue by mbkupfer** [snowflakedb/snowpark-python#462]
Please answer these questions before submitting your issue. Thanks!

1. What version of Python are you using?

Python 3.8.13 (default, Mar 28 2022, 06:59:08) [MSC v.1916 64 bit (AMD64)]

2. What operating system and processor architecture are you using?

Windows-10-10.0.19042-SP0

3. What are the component versions in the environment (`pip freeze`)?

   Replace with the output of `python -m pip freeze`
affinegap==1.12
anyio @ file:///C:/ci/anyio_1644463701441/work/dist
argon2-cffi @ file:///opt/conda/conda-bld/argon2-cffi_1645000214183/work
argon2-cffi-bindings @ file:///C:/ci/argon2-cffi-bindings_1644569878360/work
asn1crypto @ file:///tmp/build/80754af9/asn1crypto_1596577642040/work
asttokens @ file:///opt/conda/conda-bld/asttokens_1646925590279/work
attrs @ file:///opt/conda/conda-bld/attrs_1642510447205/work
Babel @ file:///tmp/build/80754af9/babel_1620871417480/work
backcall @ file:///home/ktietz/src/ci/backcall_1611930011877/work
beautifulsoup4 @ file:///C:/ci/beautifulsoup4_1650274792587/work
bleach @ file:///opt/conda/conda-bld/bleach_1641577558959/work
Bottleneck @ file:///C:/ci/bottleneck_1648029103672/work
brotlipy==0.7.0
BTrees==4.10.0
categorical-distance==1.9
certifi @ file:///C:/Windows/TEMP/abs_e9b7158a-aa56-4a5b-87b6-c00d295b01fanefpc8_o/croots/recipe/certifi_1655968940823/work/certifi
cffi @ file:///C:/ci_310/cffi_1642682485096/work
charset-normalizer @ file:///tmp/build/80754af9/charset-normalizer_1630003229654/work
cloudpickle @ file:///tmp/build/80754af9/cloudpickle_1632508026186/work
colorama @ file:///tmp/build/80754af9/colorama_1607707115595/work
cryptography @ file:///C:/ci/cryptography_1616769344312/work
cycler @ file:///tmp/build/80754af9/cycler_1637851556182/work
datetime-distance==0.1.3
debugpy @ file:///C:/ci/debugpy_1637073815078/work
decorator @ file:///opt/conda/conda-bld/decorator_1643638310831/work
dedupe==2.0.17
dedupe-variable-datetime==0.1.5
dedupe-variable-name==0.0.14
defusedxml @ file:///tmp/build/80754af9/defusedxml_1615228127516/work
DoubleMetaphone==1.1
entrypoints @ file:///C:/ci/entrypoints_1649926621247/work
executing @ file:///opt/conda/conda-bld/executing_1646925071911/work
fastjsonschema @ file:///tmp/build/80754af9/python-fastjsonschema_1620414857593/work/dist
filelock==3.8.0
fonttools==4.25.0
future==0.18.2
haversine==2.6.0
highered==0.2.1
idna @ file:///tmp/build/80754af9/idna_1637925883363/work
importlib-resources @ file:///tmp/build/80754af9/importlib_resources_1625135880749/work
ipdb==0.13.9
ipykernel @ file:///C:/ci/ipykernel_1647000978151/work/dist/ipykernel-6.9.1-py3-none-any.whl
ipympl==0.9.1
ipython @ file:///C:/ci/ipython_1651601994376/work
ipython-genutils @ file:///tmp/build/80754af9/ipython_genutils_1606773439826/work
ipywidgets @ file:///tmp/build/80754af9/ipywidgets_1634143127070/work
jedi @ file:///C:/ci/jedi_1644315425835/work
Jinja2 @ file:///opt/conda/conda-bld/jinja2_1647436528585/work
joblib @ file:///tmp/build/80754af9/joblib_1635411271373/work
json5 @ file:///tmp/build/80754af9/json5_1624432770122/work
jsonschema @ file:///C:/ci/jsonschema_1650026220037/work
jupyter==1.0.0
jupyter-client @ file:///C:/ci/jupyter_client_1650534470695/work
jupyter-console @ file:///opt/conda/conda-bld/jupyter_console_1647002188872/work
jupyter-core @ file:///C:/ci/jupyter_core_1651656285178/work
jupyter-server @ file:///C:/Windows/TEMP/abs_d3c42c59-765d-4f9b-9fa3-ad5b1369485611i_yual/croots/recipe/jupyter_server_1658754493238/work
jupyterlab @ file:///C:/ci/jupyterlab_1658909327110/work
jupyterlab-pygments @ file:///tmp/build/80754af9/jupyterlab_pygments_1601490720602/work
jupyterlab-server @ file:///C:/ci/jupyterlab_server_1650271783893/work
jupyterlab-widgets @ file:///tmp/build/80754af9/jupyterlab_widgets_1609884341231/work
kiwisolver @ file:///C:/ci/kiwisolver_1644962567532/work
Levenshtein-search==1.4.5
MarkupSafe @ file:///C:/ci/markupsafe_1621528314575/work
matplotlib @ file:///C:/ci/matplotlib-suite_1647423638658/work
matplotlib-inline @ file:///tmp/build/80754af9/matplotlib-inline_1628242447089/work
mistune==0.8.4
mkl-fft==1.3.1
mkl-random @ file:///C:/ci/mkl_random_1626186184278/work
mkl-service==2.4.0
munkres==1.1.4
nbclassic @ file:///opt/conda/conda-bld/nbclassic_1644943264176/work
nbclient @ file:///C:/ci/nbclient_1650290386732/work
nbconvert @ file:///C:/ci/nbconvert_1649741005454/work
nbformat @ file:///C:/ci/nbformat_1649845101866/work
nest-asyncio @ file:///C:/ci/nest-asyncio_1649829929372/work
notebook @ file:///C:/ci/notebook_1651648658345/work
numexpr @ file:///C:/ci/numexpr_1640704408721/work
numpy @ file:///C:/ci/numpy_and_numpy_base_1653574890766/work
opencv-python==4.6.0.66
oscrypto @ file:///tmp/build/80754af9/oscrypto_1633350059025/work
packaging @ file:///tmp/build/80754af9/packaging_1637314298585/work
pandas @ file:///C:/ci/pandas_1641443171711/work
pandocfilters @ file:///opt/conda/conda-bld/pandocfilters_1643405455980/work
parseratorvariable==0.0.18
parso @ file:///opt/conda/conda-bld/parso_1641458642106/work
patsy==0.5.2
persistent==4.9.0
pickleshare @ file:///tmp/build/80754af9/pickleshare_1606932040724/work
Pillow==9.0.1
probableparsing==0.0.1
probablepeople==0.5.4
prometheus-client @ file:///opt/conda/conda-bld/prometheus_client_1643788673601/work
prompt-toolkit @ file:///tmp/build/80754af9/prompt-toolkit_1633440160888/work
pure-eval @ file:///opt/conda/conda-bld/pure_eval_1646925070566/work
pyarrow==4.0.1
pycparser @ file:///tmp/build/80754af9/pycparser_1636541352034/work
pycryptodomex @ file:///C:/ci/pycryptodomex_1613765448814/work
Pygments @ file:///opt/conda/conda-bld/pygments_1644249106324/work
pyhacrf-datamade==0.2.6
PyJWT @ file:///C:/ci/pyjwt_1619651841937/work
PyLBFGS==0.2.0.14
pyOpenSSL @ file:///tmp/build/80754af9/pyopenssl_1635333100036/work
pyparsing @ file:///tmp/build/80754af9/pyparsing_1635766073266/work
pyrsistent @ file:///C:/ci/pyrsistent_1636111468851/work
PySocks @ file:///C:/ci/pysocks_1605287845585/work
python-crfsuite==0.9.8
python-dateutil @ file:///tmp/build/80754af9/python-dateutil_1626374649649/work
pytz==2021.3
pyusb==1.2.1
pywin32==302
pywinpty @ file:///C:/ci_310/pywinpty_1644230983541/work/target/wheels/pywinpty-2.0.2-cp38-none-win_amd64.whl
PyYAML==6.0
pyzmq @ file:///C:/ci/pyzmq_1638435185959/work
qtconsole @ file:///C:/ci/qtconsole_1657545340046/work
QtPy @ file:///opt/conda/conda-bld/qtpy_1649073884068/work
requests @ file:///opt/conda/conda-bld/requests_1641824580448/work
scikit-learn @ file:///C:/ci/scikit-learn_1642599125707/work
scipy @ file:///C:/ci/scipy_1630606942584/work
seaborn @ file:///tmp/build/80754af9/seaborn_1629307859561/work
Send2Trash @ file:///tmp/build/80754af9/send2trash_1632406701022/work
simplecosine==1.2
sip==4.19.13
six @ file:///tmp/build/80754af9/six_1644875935023/work
sklearn-pandas==2.2.0
sniffio @ file:///C:/ci/sniffio_1614030707456/work
snowflake-connector-python==2.7.12
snowflake-snowpark-python==0.9.0
soupsieve @ file:///tmp/build/80754af9/soupsieve_1636706018808/work
stack-data @ file:///opt/conda/conda-bld/stack_data_1646927590127/work
statsmodels==0.13.2
terminado @ file:///C:/ci/terminado_1644322757089/work
testpath @ file:///tmp/build/80754af9/testpath_1624638946665/work
threadpoolctl @ file:///Users/ktietz/demo/mc3/conda-bld/threadpoolctl_1629802263681/work
toml==0.10.2
tornado @ file:///C:/ci/tornado_1606942392901/work
traitlets @ file:///tmp/build/80754af9/traitlets_1636710298902/work
typing-extensions==4.3.0
urllib3 @ file:///C:/ci/urllib3_1650640043075/work
wcwidth @ file:///Users/ktietz/demo/mc3/conda-bld/wcwidth_1629357192024/work
webencodings==0.5.1
websocket-client @ file:///C:/ci/websocket-client_1614804473297/work
widgetsnbextension @ file:///C:/ci/widgetsnbextension_1645009558218/work
win-inet-pton @ file:///C:/ci/win_inet_pton_1605306167264/work
wincertstore==0.2
zipp @ file:///opt/conda/conda-bld/zipp_1641824620731/work
zope.index==5.2.0
zope.interface==5.4.0

4. What did you do?

```
dff = session.create_dataframe([1, 2, 3, 4], schema=["a"])
dff.to_pandas()
```

5. What did you expect to see?

A pandas DataFrame should be returned. Instead got the following error:
```
SnowparkFetchDataException: (1406): Failed to fetch a Pandas Dataframe. The error is: to_pandas() did not return a Pandas DataFrame. If you use session.sql(...).to_pandas(), the input query can only be a SELECT statement. Or you can use session.sql(...).collect() to get a list of Row objects for a non-SELECT statement, then convert it to a Pandas DataFrame.
```
 This wasn't an issue prior to upgrading to 0.9.0.
Now I need to do a workaround using `.collect()`, which is surprisingly seamless, but still odd how it breaks the behavior.

6. Can you set logging to DEBUG and collect the logs?

If necessary I can add this,

**Comment by sfc-gh-jdu** [comment]
Could you please upgrade your pyarrow version to 8.0.0? Snowpark 0.9.0 now requires connector 2.7.11 which requires pyarrow 8.0.0. Your pyarrow version is 4.0.1 (I'm not sure why it's not upgraded along with snowpark, but you can try and see whether it solves this issue). Thanks!

**Comment by mbkupfer** [comment]
Thanks, Upgrading pyarrow fixed it! That said, I'm on connector 2.7.12. Still works though.
